web/modules/custom/bglobal_pardot/src/Plugin/Block/PardotFormBlock.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace Drupal\bglobal_pardot\Plugin\Block;
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Session\AccountInterface;
+
+/**
+ * Build a Pardot form within a Drupal block.
+ *
+ * @Block(
+ *   id = "bglobal_pardot_block",
+ *   admin_label = @Translation("Custom Pardot form")
+ * )
+ */
+class PardotFormBlock extends BlockBase {
+
+  /**
+   * Overrides \Drupal\block\BlockPluginTrait::access().
+   */
+  public function access(AccountInterface $account, $return_as_object = FALSE) {
+    $access = AccessResult::allowedIfHasPermission($account, 'access content');
+    return $return_as_object ? $access : $access->isAllowed();
+  }
+
+  /**
+   * Build the output.
+   */
+  public function build() {
+    return [
+      '#theme' => 'bglobal_pardot_form',
+      '#action_url' => 'https://go.pardot.com/l/102272/2016-11-04/2klckm',
+      '#success_message' => 'Success!',
+      '#submit_label' => 'Act Now!',
+      '#fields' => [
```

Comment:
This `#fields` array is what controls the form presentation. We could make this configurable. Or it serves as the framework for variant forms.